Exercises/NoModules/Chapter 04/Soln4_08.cpp

```diff
@@ -0,0 +1,49 @@
+// Exercise 4-8
+// Dividing a cash amount into quarters, nickels, dimes and cents.
+
+#include <iostream>
+
+int main()
+{
+  // Declare the constants (amounts of cents)
+  const unsigned quarter {25};
+  const unsigned dime {10};
+  const unsigned nickel {5};
+
+  double amountInDollars {0.0};
+  std::cout << std::endl << "Please enter a cash amount between 0 and 10 dollars: $";
+  std::cin >> amountInDollars;
+  
+  if (amountInDollars >= 0.0 && amountInDollars <= 10.0)
+  {
+    // Multiply dollar amount by 100 ($1 {100 cents)
+    // We add 0.5 to compensate for errors in binary floating-point representation
```

Comment:
was this explained anywhere?

Reply:
Floating-point rounding issues are explained plenty of times in the first chapters, yes. Maybe not how I'd solve this today, but good enough I suppose...